From 1c3b097861ea109e54f957967811dfab344930d5 Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Thu, 16 Aug 2018 14:10:11 +0800
Subject: [PATCH 4415/5242] MLK-19228 ARM64: dts: qxp: introduce xen dtsi

commit  fe5aa250873c004d6783cdce38ebaa85571abe20 from
https://source.codeaurora.org/external/imx/linux-imx.git

xen does not support wu interrupt controller, it will cause dom0
boot failure, modify the nodes that use wu to use gic and remove wu.

Signed-off-by: Peng Fan <peng.fan@nxp.com>
Reviewed-by: Anson Huang <anson.huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qxp-mek-dom0.dts    |    9 +--
 arch/arm64/boot/dts/freescale/fsl-imx8qxp-xen.dtsi |   61 ++++++++++++++++++++
 2 files changed, 62 insertions(+), 8 deletions(-)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qxp-xen.dtsi

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-dom0.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-dom0.dts
index 88942ca..f2cf260 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-dom0.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-dom0.dts
@@ -13,7 +13,7 @@
  */
 
 #include "fsl-imx8qxp-mek.dts"
-
+#include "fsl-imx8qxp-xen.dtsi"
 
 / {
 	chosen {
@@ -28,17 +28,10 @@
                        reg = <0x00000000 0x80a00000 0x00000000 0xf93a00>;
                };
 	};
-};
 
-&lpuart0 {
-	interrupt-parent = <&gic>;
 };
 
 &imx8_gpu_ss {
 	/delete-property/ reg;
 	/delete-property/ reg-names;
 };
-
-&rtc {
-	status = "disabled";
-};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-xen.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-xen.dtsi
new file mode 100644
index 0000000..2d600d0
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-xen.dtsi
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2018 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+/ {
+	/delete-node/ wu;
+};
+
+&usbotg1 {
+	interrupt-parent = <&gic>;
+};
+
+&flexcan1 {
+	interrupt-parent = <&gic>;
+};
+
+&flexcan2 {
+	interrupt-parent = <&gic>;
+};
+
+&flexcan3 {
+	interrupt-parent = <&gic>;
+};
+
+&usbotg3 {
+	interrupt-parent = <&gic>;
+};
+
+&lpuart0 {
+	interrupt-parent = <&gic>;
+};
+
+&lpuart1 {
+	interrupt-parent = <&gic>;
+};
+
+&lpuart2 {
+	interrupt-parent = <&gic>;
+};
+
+&lpuart3 {
+	interrupt-parent = <&gic>;
+};
+
+&fec1 {
+	interrupt-parent = <&gic>;
+};
+
+&fec2 {
+	interrupt-parent = <&gic>;
+};
-- 
1.7.9.5

